src/cray/boa/capmcclient.py

```diff
@@ -192,6 +199,10 @@ def power(nodes, state, force=True, session=None, reason="BOA: Powering nodes"):
 
     session = session or requests_retry_session()
     prefix, output_format = node_type(nodes)
+    if output_format == 'nids':
+        raise CapmcDeprecationException("CAPMC deprecated power control for nid based entries; "
```

Comment:
The 'boot' function in the agent code at line 772 calls this 'power' function. 'boot' doesn't expect to raise this particular exception nor does it explicitly catch it.
The docstring says, "        Raises:
          ServiceNotReady -- If it fails to interact with BSS
          NontransientException -- If nodes were not ready."
You should update the docstring at a minimum.
Did you boot a node using NIDs to see what happened?

Reply:
Done.

Reply:
The expected behavior is for the exception to raise to the top of the agent code so that the user can make a recommended set of features in their session template. Without this specific exception, a call would be made to capmc that would _look_ like capmc failed with a 4XX error; this is important as HTTP errors here are parsed by the parse_response function, which treats the request to the client as something transient that should be re-attempted.

In keeping with the laid out concept of transient versus non-transient, the deprecation class should be considered non-transient so that the agent fully stops. In testing, this provided a cleaner exit strategy while using agent code, and I'll update the base class of the exception to reflect this.